nemo/collections/tts/torch/data.py

```diff
@@ -980,3 +1072,319 @@ def __getitem__(self, index):
+                Also used if speaker stats are not available for the given speaker
+            pitch_std (Optional[float]): If using global normalization, normalize using these statistics. 
+                Also used if speaker stats are not available for the given speaker
+            pitch_normalization (str): Can be one of ['speaker_wise', 'global', 'none']. Indicates the kind of pitch normalization.
+            sup_data_dir (Optional[Union[str, Path]]): Data directory containing pre-computed embeddings/statistics. If set as 
+            speaker_stats_pitch_fp (Optional[Union[str, Path]]): Path to the json containing speaker pitch stats. 
+                If set as None, tries to lookup for a default filename (speaker_pitch_stats.json) in sup_data_dir. 
+                Needed if we use pitch_normalization is "speaker_wise"
+            speaker_conditioning_type (Optional[str]): Can be one of ["per_sample", "mean", "interpolate"]. Defaults to "per_sample"
+                per_sample: Speaker embedding computed from the same utterance
+                mean: Speaker embedding for all utterances of a given speaker is the same and equal to the mean speaker embedding. 
+                interpolate: Interpolate b/w per_sample and mean speaker embedding.
+        """
+        assert ssl_content_emb_type in ["probs", "embedding", "log_probs", "embedding_and_probs"]
+
+        if isinstance(manifest_filepath, str):
+            manifest_filepath = [manifest_filepath]
+        self.manifest_filepath = manifest_filepath
+
+        data = []
+        total_duration = 0
+        for manifest_file in self.manifest_filepath:
+            with open(Path(manifest_file).expanduser(), 'r') as f:
+                logging.info(f"Loading dataset from {manifest_file}.")
+                for line in tqdm(f):
+                    item = json.loads(line)
+                    if "speaker" not in item:
+                        item["speaker"] = 0
+                    file_info = {
+                        "audio_filepath": item["audio_filepath"],
+                        "duration": item["duration"] if "duration" in item else None,
+                        "speaker": item["speaker"] if "speaker" in item else 0,
+                        "dataset_id": item["dataset_id"] if "dataset_id" in item else 0,
+                    }
+
+                    data.append(file_info)
+
+                    if file_info["duration"] is None:
+                        logging.info(
+                            "Not all audio files have duration information. Duration logging will be disabled."
+                        )
+                        total_duration = None
+
+                    if total_duration is not None:
+                        total_duration += item["duration"]
+
+        logging.info(f"Loaded dataset with {len(data)} files.")
+        if total_duration is not None:
+            logging.info(f"Dataset contains {total_duration / 3600:.2f} hours.")
```

Comment:
At some point we should separate this into a separate function that can be shared between TTS data loading classes. Don't have to do it in this PR, just putting this down as a TODO.

Reply:
added a todo